### PR TITLE
Spec Terms

### PR DESCRIPTION
**Pull request by csarven**
Spec Terms provides classes and properties that can be used to describe any significant unit of information in technical reports.

Influenced by a number of works encompassing W3C processes, specification authoring and publishing, quality assurance, e.g.:
* https://www.w3.org/Consortium/Process/
* https://www.w3.org/TR/spec-variability/
* https://www.w3.org/TR/qaframe-spec/
* https://www.w3.org/TR/test-metadata/
* https://www.w3.org/TR/EARL10-Schema/

It is currently used by Solid Technical Reports, conformance test suite coverage documents, and applications e.g.:
* https://solidproject.org/TR/protocol (see https://solidproject.org/ED/protocol for most current use.)
* https://solidproject.org/TR/wac
* https://solidproject.org/TR/notifications-protocol
* https://solid-contrib.github.io/specification-tests/coverage
* https://dokie.li/

The work here fits within the scope of https://github.com/solid/process/blob/main/test-suite-panel-charter.md , W3C QA activities, and the practice (the art?) of writing technical reports.

The work here is far from "complete" and there is much to look ahead. Consumers of the vocabulary should note that it currently has the status of "testing". The work here is mature enough to open up further, i.e., merge to main branch and deploy changes regularly under the designated namespace, and gather more implementation experience.

This is also a call to gather feedback to improve the vocabulary design; to encourage its use in publication of technical reports; to encourage the consumption of machine-readable technical reports by exciting applications.

### CODE REVIEW

**Review by edwardsph — APPROVED** [review]
LGTM with a couple of comments.
Some properties include owl classes e.g. `owl:ObjectProperty` but not all - should this be consistent?

**Comment by csarven** [comment]
I agree it should be consistent or clarified as much as possible. Various things are under specified and they'll need to be reviewed.

**Review by michielbdejong — APPROVED** [review]

File: spec.ttl [hunk 1]
```diff
@@ -0,0 +1,595 @@
...
+@prefix test-description: <http://www.w3.org/2006/03/test-description#> .
+@prefix vann: <http://purl.org/vocab/vann/> .
+@prefix vs: <http://www.w3.org/2003/06/sw-vocab-status/ns#> .
+@prefix xsd: <http://www.w3.org/2001/XMLSchema#> .
+
+
+spec:
+  a owl:Ontology ;
+  rdfs:label "Spec Terms"@en ;
+  dcterms:description "Spec Terms provides classes and properties that can be used to describe any significant unit of information in technical reports."@en ;
```
Comment (TallTed) [inline spec.ttl:18]:
```suggestion
  rdfs:label "Spec Terms ontology"@en ;
  dcterms:description "The Spec Terms ontology provides classes and properties that can be used to describe any significant unit of information in technical reports."@en ;
```

File: spec.ttl [hunk 2]
```diff
@@ -0,0 +1,595 @@
...
+spec:InternationalizationConsideration
+  a rdfs:Class ;
+  rdfs:subClassOf spec:Consideration ;
+  rdfs:label "Internationalization Consideration"@en .
+
+spec:RegistrationConsideration
+  a rdfs:Class ;
+  rdfs:subClassOf spec:Consideration ;
+  rdfs:label "Registration Consideration"@en ;
+  rdfs:comment "Registration considerations at standards bodies eg. Registering HTTP Link Relations at IANA."@en .
```
Comment (TallTed) [inline spec.ttl:332]:
```suggestion
  rdfs:comment "Registration considerations at standards bodies, e.g., Registering HTTP Link Relations at IANA."@en .
```

File: spec.ttl (hunk 1 above)
Comment (csarven) [reply spec.ttl:18]: Thank Ted. I say the following without wishing us getting drawn into a discussion on whether what's herein is a vocabulary or an ontology.

As I see it, Spec Terms is not detailed or complex enough or intended to be an ontology in its proper sense. It is intended as a simple vocabulary. I'm well-aware that the document includes OWL (but primarily RDFS and SKOS) however don't think that the human-readable content should express that.

Would you object to not mentioning "ontology" in the human-readable text?

**Comment by csarven** [comment]
@karlcow , hi! :)

**Comment by csarven** [comment]
@edwardsph , I've added owl:ObjectProperty to some properties in https://github.com/solid/vocab/pull/84/commits/cd81f14fa2642d844ac972c171e0f0a67712767e . If things break, it is your fault.

**Review by kjetilk — COMMENTED** [review]
Looks good. I'd just like to make one little request, if I may:

Can we please use `_` instead of `-` in terms? 

In many programming languages, dashes can't be used in methods and variable names, but underscores can, so using underscores leads to simpler programming when using a translator between terms and programming languages.

**Comment by matthieubosquet** [comment]
@kjetilk’s comment makes me think: I’m not sure why we would use a different casing convention and - for those terms.

Couldn’t we simply keep all terms camel case (classes/concepts/named individuals upper and properties lower)?

**Comment by csarven** [comment]
I won't bother diving into the template or readability that I applied or try to convince everyone here. Open to whatever convention but I'll just say that simple patterns/consistency is not necessarily ideal in the end.

Having said that, I'm okay with convention camelCase and UpperCase. The property/class name is generally intended to be derived from the natural language label, e.g., the "Producer of content" concept becomes `ProducerOfContent`.

I'll propose example changes here and if we can get a few +1/-1, I can update the PR:

* `MUST-NOT` -> `MUSTNOT` -- dropping `-` and keeping this all uppercase because of RFC 8173. Similar terms where case sensitivity is meaningful will be retained and not changed for the purpose of the vocab.
* `strongly-encouraged` -> `StronglyEncouraged` -- dropping `-` and going with upper.

If there are other ones issues or preferences, let's get them in. Thanks.

**Comment by matthieubosquet** [comment]
> I'll propose example changes here and if we can get a few +1/-1, I can update the PR:
>     * `MUST-NOT` -> `MUSTNOT` -- dropping `-` and keeping this all uppercase because of RFC 8173. Similar terms where case sensitivity is meaningful will be retained and not changed for the purpose of the vocab.
>     * `strongly-encouraged` -> `StronglyEncouraged` -- dropping `-` and going with TitleCase.
>     * "Producer of content" from label -> `ProducerOfContent` concept -- that's what's used at the moment, but `ProducerContent` seems okay too - does anyone have opinions on this?

+1 on all of those.

**Review by matthieubosquet — APPROVED** [review]
LGTM pending [suggestions](https://github.com/solid/vocab/pull/84#issuecomment-1316746637).

File: spec.ttl [hunk 3]
```diff
@@ -0,0 +1,595 @@
...
+  a skos:Concept ;
+  skos:prefLabel "Protocol"@en ;
+  skos:definition "Describes the interaction between two parties. Example: SOAP [SOAP]."@en .
+
+spec:ProcessorBehaviour
+  a skos:Concept ;
+  skos:prefLabel "Processor behaviour"@en ;
+  skos:definition "Describes the behavior of a piece of software that takes well-defined inputs and operates on them to produce specified output. Example: the XSL Transformations language [XSLT] specifies the behavior of an XSLT processor. The specification describes an XML-based style sheet language that directs the operations of a processor, but it clearly states that conformance determinations apply to the processor, not the style sheets. Example: XML Query specifies the behavior of a processor that takes the query language and fetches data, typically arranging that data in a convenient form for XML/Web use. It allows flexibility of the storage system from which data is fetched."@en .
+
+spec:APIs
```
Comment (matthieubosquet) [inline spec.ttl:501]: Why plural?

Comment (csarven) [reply spec.ttl:501]: The term is defined in specification categories: https://www.w3.org/TR/spec-variability/#spec-cat (:cat2:). I originally had API then changed to APIs to stay close to the original term. I suppose a spec could offer multiple APIs? I don't have a strong opinion on this.

**Review by RubenVerborgh — APPROVED** [review]

File: spec.ttl (hunk 3 above)
Comment (matthieubosquet) [reply spec.ttl:501]: Oh yeah ^^, I see it's inconsistent with line 463 actually `spec:API`.

An API spec sounds and feels better to me than an APIs spec. I'm generally prejudiced against plural. But yeah I can live with either.

**Pull request merged**